pkg/operator/vsphere_check.go

```diff
@@ -376,3 +379,13 @@ func newClient(ctx context.Context, cfg *vsphere.VSphereConfig, username, passwo
 
 	return client, restClient, nil
 }
+
+// isValidvCenterUsernameWithDomain checks whether username is valid or not.
+// vSphere considers the username invalid if it does not contain a fully qualified domain name.
+// This function was taken from the CSI driver code:
+// Take from https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/a7e37ef11bad80693a250ab5a815b527d99e3401/pkg/common/config/config.go#L326-L336
+func isValidvCenterUsernameWithDomain(username string) bool {
+	regex := `^(?:[a-zA-Z0-9.-]+\\[a-zA-Z0-9._-]+|[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+)$`
```

Comment:
Can we also log the previous message? The idea is - while a log will be easily visible in must-gather, we won't be able to see the metric unless customer specifically mentions it.